specification/security/resource-manager/Microsoft.Security/preview/2019-01-01-preview/networkData.json

```diff
@@ -0,0 +1,361 @@
+            }
+          }
+        },
+        "actualTraffic": {
+          "readOnly": true,
+          "type": "object",
+          "description": "The actual traffic details",
+          "properties": {
+            "startTime": {
+              "type": "string",
+              "format": "date-time",
+              "readOnly": true,
+              "description": "The UTC start time of the actual traffic time range"
+            },
+            "endTime": {
+              "type": "string",
+              "format": "date-time",
+              "readOnly": true,
+              "description": "The UTC end time of the actual traffic time range"
+            },
+            "trafficDataState": {
+              "type": "string",
+              "description": "Indicate whether the Azure monitor agent is installed on the vm",
+              "enum": [
+                "Available",
+                "Not available"
+              ],
+              "x-ms-enum": {
+                "name": "trafficDataState",
+                "modelAsString": true,
+                "values": [
+                  {
+                    "value": "Available",
+                    "description": "The Azure monitor agent is installed on the vm"
+                  },
+                  {
+                    "value": "Not available",
+                    "description": "the Azure monitor agent is not installed on the vm"
+                  }
+                ]
+              }
+            },
+            "inboundConnectedResources": {
+              "$ref": "#/definitions/ActualTrafficConnectedResources"
+            },
+            "outboundConnectedResources": {
+              "$ref": "#/definitions/ActualTrafficConnectedResources"
+            }
+          }
+        },
+        "unusedPorts": {
+          "readOnly": true,
+          "type": "object",
+          "description": "The unused ports details",
+          "properties": {
+            "inboundConnectedResources": {
+              "$ref": "#/definitions/ConnectedResources"
+            },
+            "outboundConnectedResources": {
+              "$ref": "#/definitions/ConnectedResources"
+            }
+          }
+        }
+      }
+    },
+    "ConnectedResources": {
+      "readOnly": true,
+      "type": "object",
+      "description": "The connections details",
+      "properties": {
+        "tcpPortsCount": {
+          "readOnly": true,
+          "type": "integer",
+          "description": "The allowed connections tcp ports count"
+        },
+        "udpPortsCount": {
+          "readOnly": true,
+          "type": "integer",
+          "description": "The allowed connections udp ports count"
+        },
+        "tcpPorts": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The allowed connections tcp ports"
+        },
+        "udpPorts": {
+          "readOnly": true,
+          "type": "string",
+          "description": "The allowed connections udp ports"
```

Comment:
the examples show these as an array inside a string.  Why not have an array of strings?

Reply:
Currently this is the convention is use and similar to other API's we work with / expose.

Reply:
What is the expectation here?  The user parses the string as JSON and then consumes it as an array?  How else are they reasonably supposed to consume this?  Microsoft.Network related APIs set the standard for these types of semantics and you should attempt to follow similar conventions when representing similar data (i.e. destinationPortRanges in https://docs.microsoft.com/en-us/rest/api/virtualnetwork/securityrules/createorupdate).

Reply:
2 reasons here:
1. We already have this format in other Microsoft.Security Rest and our customers familiar with it. 
2. This API serves as a way for the user to see its security status (from network segmentation perspective) - The most common use is to fetch data for the entire subscription to see which resources are open and which are not.
Having too much details will results is huge collections.

Reply:
If its in your APIs already then for internal consistency I guess its fine